src/index.ts

```diff
@@ -46,51 +47,122 @@ declare global {
 
 /**
- * Removes all listeners added in a certain mode.
+ * Calls the respective keypress function depending on current mode.
+ * @param event The event of a keypress
  */
-function removeListeners() {
-    if (modeState === "ellipseMode") {
-        canvas.removeEventListener("mousedown", cutHandler);
-        canvas.removeEventListener("mousemove", cutHandler);
-        canvas.removeEventListener("mouseup", cutHandler);
-        canvas.removeEventListener("mouseout", cutHandler);
-    } else if (modeState === "atomMode") {
-        window.removeEventListener("keypress", atomHandler);
-        canvas.removeEventListener("mousedown", atomHandler);
-        canvas.removeEventListener("mousemove", atomHandler);
-        canvas.removeEventListener("mouseup", atomHandler);
-        canvas.removeEventListener("mouseout", atomHandler);
+function keyPressHandler(event: KeyboardEvent) {
+    switch (modeState) {
+        case "atomMode":
+            atomKeyPress(event);
+            break;
+    }
+}
+
+/**
+ * Calls the respective mouse down function depending on current mode.
+ * @param event The mouse down event
+ */
+function mouseDownHandler(event: MouseEvent) {
+    switch (modeState) {
+        case "cutMode":
+            cutMouseDown(event);
+            hasMouseDown = true;
+            break;
+        case "atomMode":
+            hasMouseDown = atomMouseDown(event);
+            break;
+    }
+}
+
+/**
+ * If mouse down has been used calls the respective mousedown function based on mode.
+ * @param event the mouse move event
+ */
+function mouseMoveHandler(event: MouseEvent) {
+    switch (modeState) {
+        case "cutMode":
+            if (hasMouseDown) {
+                cutMouseMove(event);
+            }
+            break;
+        case "atomMode":
+            if (hasMouseDown) {
+                atomMouseMove(event);
+            }
+            break;
+    }
+}
+
+/**
+ * If mouse down has been used calls the respective mouse up function based on mode.
+ * Resets mouse down after use.
+ * @param event The mouse up event
+ */
+function mouseUpHandler(event: MouseEvent) {
+    switch (modeState) {
+        case "cutMode":
+            if (hasMouseDown) {
+                cutMouseUp(event);
+            }
+            hasMouseDown = false;
+            break;
+        case "atomMode":
+            if (hasMouseDown) {
+                atomMouseUp(event);
+            }
+            hasMouseDown = false;
+            break;
+    }
+}
+
+/**
+ * If mouse down has been used calls the respective mouse out function based on mode.
+ * Resets mouse down after use.
+ */
+function mouseOutHandler() {
+    switch (modeState) {
+        case "cutMode":
+            if (hasMouseDown) {
+                cutMouseOut();
+            }
+            hasMouseDown = false;
+            break;
+        case "atomMode":
+            if (hasMouseDown) {
+                atomMouseOut();
+            }
+            hasMouseDown = false;
+            break;
```

Comment:
The mouse is still down even if the mouse is outside of the canvas, if you come back in while the mouse is down that's additional logic for the individual mode handlers, you could additional global Boolean for mouseIn to handle this case.